sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -609,6 +609,20 @@ private[hive] class HiveClientImpl(
     shim.alterTable(client, qualifiedTableName, hiveTable)
   }
 
+  override def alterTableStats(
+      dbName: String,
+      tableName: String,
+      stats: Map[String, String]): Unit = withHiveState {
+    val hiveTable = getRawHiveTable(dbName, tableName).rawTable.asInstanceOf[HiveTable]
+    val newParameters = new JHashMap[String, String]()
+    hiveTable.getParameters.asScala.toMap.filterNot(_._1.startsWith(STATISTICS_PREFIX))
```

Comment:
It's a bit tricky to make `HiveClient` handle this `STATISTICS_PREFIX`. It should be the responsibility of `HiveExternalCatalog`. `HiveClient` should only take care of the communication with HMS.